From 941886c8d4ca7655cdd165eeb439d93abcf9eab2 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 18 Sep 2016 14:01:14 +0200
Subject: [PATCH 07/33] Prevent debug library names in pkg-config files

qmake generates the pkgconfig .pc files two times, once for the
release build and once for the debug build (which we're not actually
building in this package). For both generations the exact same
pkgconfig file name is used. This causes references to the debug
build ending up in the .pc files which are unwanted
Prevent this from happening by giving the pkgconfig .pc
files for the debug build an unique file name.
---
 qmake/generators/makefile.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/qmake/generators/makefile.cpp b/qmake/generators/makefile.cpp
index 5abf0624f4..a7f1285c26 100644
--- a/qmake/generators/makefile.cpp
+++ b/qmake/generators/makefile.cpp
@@ -3203,6 +3203,9 @@ MakefileGenerator::pkgConfigFileName(bool fixify, bool onlyPrependDestdir)
         if (dot != -1)
             ret = ret.left(dot);
     }
+    if (project->isActiveConfig("debug")) {
+        ret += "d";
+    }
     ret += Option::pkgcfg_ext;
     QString subdir = project->first("QMAKE_PKGCONFIG_DESTDIR").toQString();
     if(!subdir.isEmpty()) {
-- 
2.19.2

